### ISSUE TITLE
Cannot retrieve keyring password from non-default storage

### ISSUE THREAD

**Issue by cbop-dev** [gilbertchen/duplicacy#184]
Having read and applied the patch from [issue #103](https://github.com/gilbertchen/duplicacy/issues/103) involving storing keyring passwords, I could not get the command-line tool to actually read the stored password for a _non-default storage_ (using the -storage parameter), even though the keyring password was correctly added. E.g., calling `duplicacy -d list -storage [storagename]` yields a `Failed to get the value from the keyring: keyring: Password not found.` and a password prompt, even after having stored the keyring.

The reason, as far as I can tell by troubleshooting (using the Linux commands secret-tool and  debug_monitor) and looking at duplicacy_utils.go, is that while the ["SavePassword" method](https://github.com/gilbertchen/duplicacy/blob/master/src/duplicacy_utils.go#L195) correctly checks for a non-default storage parameter, and then sets the password to "[storage name]_password", the [GetPassword function](https://github.com/gilbertchen/duplicacy/blob/master/src/duplicacy_utils.go#L153) does not. Thus, even when passing a `-storage` parameter, the program looks for a "password" attribute from the keyring instead of "[storagename]_password"  attribute which the SavePassword method would have used. 

To fix this, simply I added the following code in [duplicacy_utils.go](https://github.com/gilbertchen/duplicacy/blob/master/src/duplicacy_utils.go#L164) at line 164 in my own installation, recompiled, and everything worked dandy: 

    if preference.Name != "default" {
        passwordID = preference.Name + "_" + passwordID
    }

Perhaps this fix should be added to the master branch?

**Comment by gilbertchen** [comment]
Yes, that is a bug introduced by a previous change.  I just checked in https://github.com/gilbertchen/duplicacy/commit/d1817ae5573499105f469a1d386c21012b9c528e which is exactly what you suggested here.

**Comment by Quixocrates** [comment]
I still get this error in Duplicacy on my Mac Server. How do I add those lines of code to my build?

**Comment by gilbertchen** [comment]
You can run `duplicacy -d list` to see if Duplicacy is reading from the correct environment variable for the password.
